### ISSUE TITLE
Regression with subsites compatibility

### ISSUE THREAD

**Issue by robbieaverill** [silverstripe/silverstripe-securityreport#38]
There's a test failing against the latest subsites version: https://travis-ci.org/silverstripe/cwp-recipe-kitchen-sink/jobs/423801729#L950

This would've been caused by https://github.com/silverstripe/silverstripe-subsites/pull/388

Needs investigation. We should start by separating that test into individual cases to ensure it's not pollution of the global state between loading each fixture and running tests on them

**Comment by robbieaverill** [comment]
I've confirmed it as a regression, not just in the tests

**Comment by robbieaverill** [comment]
I've made a mistake in my thinking with that subsites pull request. In psuedo code, this is what I've changed it to do when you're asking which subsites can be accessed for a given permission code:

* Loop all subsites
* Set current subsite to each
* Ask Permission::checkMember whether the code is valid in the current state

I think the problem is that the permission check doesn't cross reference against the subsites a user's group can be used in, so they're all treated as global checks.

In practice the permission system seems to be working, but the API has clearly regressed since this report is broken.

I'll need to reintroduce some of the previous logic to run SQL queries against the member's groups in order to filter permissions available for each group the member is in, which is in turn filtered against the current subsite.

The Permission::checkMember() method won't work I don't think =(
